uniffi/src/bindings/kotlin/templates/macros.kt

```diff
@@ -5,17 +5,21 @@
 #}
 
 {%- macro to_rs_call(func) -%}
-_UniFFILib.INSTANCE.{{ func.ffi_func().name() }}({% call _arg_list_rs_call(func.arguments()) -%})
+ rustCallWith{% match func.ffi_func().throws() %}{% when Some with (e) %}{{e}}{% else %}{% endmatch %} { e -> 
```

Comment:
How do you feel about moving the `match` up one level, something like:

```text
{% match func.ffi_func().throws() %}
{% when Some with (e) %}
rustCallWith{{e}} { e -> ... }
{% else %}
rustCallWith { e -> ...}
{% endmatch %}
```

(Also, did I read that right? I don't see a `rustCallWith` yet, is that for default errors, or something else? Maybe I should just keep reading 😅) I know it's a couple more duplicated lines, but the inline `match` took me a couple tries to parse on first read!

Reply:
Ya the `rustCallWith` stuff is an ugly bi-product of me not knowing how to do generics well in Swift and Kotlin and thinking "let me just get it to work". 

I'll change those to be just a `rustCall` and will do some generics magic ✨  to make it work.

But you are right, `rustCallWith` (without any suffix) doesn't exist, but since all functions generated using this take errors, the `else` won't ever happen. It's very ugly I know 😅 Hopefully will be better once I add the generics and simplify the errors to one error.